### PR TITLE
Fixed Patient Signup and Deletion from the database 

### LINKED ISSUE #53: Implement Signin and Signup of the patient.
I can send the form the data while adding new patients to the backend and data will be saved in the database.
I will require
using dotenv library to fix CORS policy
Send the data to backend
Data to be saved in database

### PR DESCRIPTION
**Pull request by sujal1256**
Fixes #53 

Created a route for registering patient
Created a route for getting all patients
Created a route for deleting a patient
and controllers for all of them

image

Take a look at the video itself, now you will be able to create and remove patients and everything will be updated in the database
https://github.com/user-attachments/assets/159a70e1-272c-4f67-bf8d-05e92bd9621e

### CODE REVIEW

**Comment by sujal1256** [comment]
Hey @yazdanhaider, I don't know the reason why a test has not passed please let me know what should I do to fix this?

**Comment by sujal1256** [comment]
Hey @yazdanhaider, Can you please tell me if this PR is okay, or is there any other commit for this PR to be accepted.

File: Backend/api/controllers/PatientController.js [hunk 1]
```diff
@@ -1,39 +1,45 @@
...
-import jwt from 'jsonwebtoken';
+import { Patient } from "../models/Patient.js";
+import bcrypt from "bcrypt";
+import jwt from "jsonwebtoken";
 
 // Register a new patient
 export const registerPatient = async (req, res) => {
   try {
-    const { name, email, gender, mobile, password } = req.body;
+    const { name, email, gender, mobile, password, age} = req.body;
```
Comment (yazdanhaider) [inline Backend/api/controllers/PatientController.js:9]: Remove the password field when adding a new patient. Replace the password field with an ID Number field instead.

File: Frontend/src/components/Patients.jsx [hunk 2]
```diff
@@ -1,167 +1,226 @@
...
+    },
+    {
+      id: 2,
+      name: "Jane Smith",
+      age: 28,
+      gender: "Female",
+      contact: "987-654-3210",
+    },
+  ]);
+  const [searchTerm, setSearchTerm] = useState("");
```
Comment (yazdanhaider) [inline Frontend/src/components/Patients.jsx:23]: There will be a alert message when the patient name is not upto 4 characters. 
alert('Invalid name. Name should start with a letter and be at least 4 characters long.

File: Frontend/src/components/Patients.jsx [hunk 3]
```diff
@@ -1,167 +1,226 @@
...
+  const handleInputChange = (e) => {
+    setNewPatient({ ...newPatient, [e.target.name]: e.target.value });
+  };
 
-    const validateForm = () => {
-        const nameRegex = /^[A-Za-z][A-Za-z0-9 ]{3,}$/;
-        const contactRegex = /^[0-9]{3}-[0-9]{3}-[0-9]{4}$/;
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    try {
```
Comment (yazdanhaider) [inline Frontend/src/components/Patients.jsx:40]: Also alert for invalid formatting for mobile number.
alert('Invalid contact number. It should follow the format XXX-XXX-XXXX.'

**Review by yazdanhaider — CHANGES_REQUESTED** [review]
Heyyy @sujal1256 Please fix the small bugs and resolve the error occurring while adding a patient. Also ensure proper error handling implementation throughout the process.

File: Backend/api/controllers/PatientController.js (hunk 1 above)
Comment (sujal1256) [reply Backend/api/controllers/PatientController.js:9]: So how do you want to create the Id for the parent?

Comment (sujal1256) [reply Backend/api/controllers/PatientController.js:9]: Please tell me this as I also want to work on some other issue in your project.

Comment (yazdanhaider) [reply Backend/api/controllers/PatientController.js:9]: @sujal1256 This is the admin panel and we will be creating a new page specifically for adding patients. In the admin panel, patients will be added using their ID number not by their password.

Comment (sujal1256) [reply Backend/api/controllers/PatientController.js:9]: Ok so I am creating a unique Id for the patient using a npm library, and there will no password comming from frontend.

**Comment by sujal1256** [comment]
I just updated my fork and pulled it inside my repo,
and I was counted as a commit.
I hope it is fine

**Comment by yazdanhaider** [comment]
Heyy @sujal1256 The “ID Number” text field is missing. Please add/fix it.

Screenshot 2024-10-22 at 2 16 34 AM

**Comment by sujal1256** [comment]
@yazdanhaider, the ID number is created in backend using UUID library, do you want the user to enter the ID?
It will collapse with other patient's id

**Comment by sujal1256** [comment]
Hey @yazdanhaider, As you said now, user will be sending the patient Id and it will saved in the database,
If this is fine with you, should I make the commit.

**Pull request merged**